.travis.yml

```diff
@@ -40,6 +40,11 @@ deploy:
     on:
       branch: master
       jdk: openjdk8
+  - provider: releases
+    api_key: "${GITHUB_WRITE_TOKEN}"
```

Comment:
Don't we also need `skip_cleanup: true`?

Reply:
Only if you want to publish additional artifacts, but then this would also need `file: ...`. The source code archives (see e.g. [release of v0.1.2.](https://github.com/junit-pioneer/junit-pioneer/releases/tag/v0.1.2)) are added per default.

Reply:
Here are two examples from our repos:

1. Publish drafts (i.e. someone still has to manually confirm the release) without additional artifacts:
    * [`.travis.yml`](https://github.com/retest/recheck/blob/master/.travis.yml#L34)
    * [resulting releases](https://github.com/retest/recheck/releases)
2. Publish drafts with an additional fat JAR (needs `skip_cleanup` and `file`):
    * [`.travis.yml`](https://github.com/retest/recheck-web/blob/master/.travis.yml#L41)
    * [resulting releases](https://github.com/retest/recheck-web/releases)

Reply:
So you're saying the release plugin knows what files to add to the release? That's mildly surprising to me and [the documentation](https://docs.travis-ci.com/user/deployment/releases/#uploading-multiple-files) doesn't seem to mention it, but since it seems to work for Recheck, we can give it a try. :grin:

Reply:
Magic! 🧙‍♂️

But no, it doesn't know what files to upload. Travis/GitHub only automatically archives (as zip and tar.gz) the source code of the release commit and adds this to the assets.

If there is nothing additionally to upload, there is no need to use `file` and `skip_cleanup`. Or is there anything you would like to add, in contrast to e.g. release v0.1.2?

Reply:
Oh wait, so this is just a ZIP/TAR of the entire repo at that point in time? Well, that's useless isn't it? You can get the same by navigating to [a tag](https://github.com/junit-pioneer/junit-pioneer/tree/v0.3.2) and hit download there. No, if we do this, we should do it right:

* include library JAR
* include sources JAR
* include Javadoc JAR

We could even go further and try to [pass a name and a body](https://docs.travis-ci.com/user/deployment/releases/#advanced-options), but that will be trickier.

I understand that this is more work than you were planning for. If you don't want to do this, that's perfectly ok, just let me know.

Reply:
No problem, I can take care of it. I just wonder: Would anyone actually use Pioneer _without_ a build tool / something that takes care of dependency management, i.e., download the JARs from here, add the missing dependencies manually and then use it?

Reply:
Idk, I wouldn't. :grin: But if not, then why go through the trouble of dealing with GitHub's releases at all?

Reply:
That's a valid question and I've asked this myself too some time ago. Let me show you this piece of art:

![grafik](https://user-images.githubusercontent.com/11643714/67275814-aecd4900-f4c3-11e9-9f51-2bc491ef9351.png)

😉

I'd say it is a bit more user-friendly to present releases like this, but overall just a matter of taste.

However, as you already pointed out, we can also leave it as it is?

Reply:
> However, as you already pointed out, we can also leave it as it is?

Nah, no I want to do it. But right, with bells and whistles! :blush:

Reply:
I'm a Gradle noob, but I assume `"./build/libs/junit-pioneer-*.jar"` would include the library, sources and Javadoc JARs? At least that's my impression after doing `gradle assemble` locally.

Maybe we have to trigger assembly in `before_deploy` to make sure the JARs are actually there. Could you please verify this? For example, create a tag with this release provider being active, then we should see what gets uploaded to GitHub. (You can also add `draft: true` for testing purpose, then the release isn't automatically published.)

Reply:
Ack on first paragraph.

Re second: If I get the docs correctly, I'd add `skip_cleanup: true` and hope that it means the previous build step doesn't get cleaned up, so we can reuse its artifacts.

Reply:
Oh, it's already there, nice. To be safe, lets move the `release` blog to line 37 (i.e. before triggering the website build), so it can reuse the same artifacts.